libyul/optimiser/CommonSubexpressionEliminator.cpp

```diff
@@ -52,6 +53,16 @@ CommonSubexpressionEliminator::CommonSubexpressionEliminator(
 {
 }
 
+void CommonSubexpressionEliminator::operator()(FunctionDefinition& _fun)
+{
+	ScopedSaveAndRestore returnVariables(m_returnVariables, {});
```

Comment:
```suggestion
	m_returnVariables = {};
```
Maybe more intuitive? Might need to add FunctionHoister to the prerequisites.

Reply:
It's safer this way.